tests/rptest/tests/controller_snapshot_test.py

```diff
@@ -114,14 +114,30 @@ def _check_features(self, other):
                 f"features map mismatch (feature: {f}): got {new_v}, expected {v}"
 
     def _check_config(self, other):
-        def to_set(resp):
-            return set((k, (v if not isinstance(v, list) else tuple(v)))
-                       for k, v in resp.items())
-
-        symdiff = to_set(self.config_response) ^ to_set(other.config_response)
+        # The elements of constraints are dicts which are not hashable and can
+        # not be in a set. So we compare constraints on their own. Python will
+        # compare the elements at each index and comparing dicts checks the
+        # key and value.
+        def to_set_without_constraints(resp):
+            config_set = set()
+            for k, v in resp.items():
+                if k != 'constraints':
+                    config_set.add(
+                        (k, (v if not isinstance(v, list) else tuple(v))))
+            return config_set
+
+        symdiff = to_set_without_constraints(
+            self.config_response) ^ to_set_without_constraints(
+                other.config_response)
         assert len(
             symdiff) == 0, f"config responses differ, symdiff: {symdiff}"
 
+        self_constraints = self.config_response['constraints']
+        self_constraints.sort(key=lambda con: con['name'])
+        other_constraints = other.config_response['constraints']
+        other_constraints.sort(key=lambda con: con['name'])
+        assert self_constraints == other_constraints, f"{self_constraints} != {other_constraints}"
+
```

Comment:
@ztlpn Could you take a look at this? The change is minimal but it edits a controller snapshots test.

Here is a summary on the constraints property: Constraints is a sequence of YAML objects that enable the broker to reject topics that do not satisfy the constraint **or** clamp topic properties to cluster-level properties. The YAML objects indicate the: name of the cluster-level config to focus on, the type of constraint (either restrict or clamp), and the values for the constraint (min/max for integrals, a boolean for non-integrals).

Reply:
Moving this to a separate ticket and PR. https://github.com/redpanda-data/redpanda/issues/14018